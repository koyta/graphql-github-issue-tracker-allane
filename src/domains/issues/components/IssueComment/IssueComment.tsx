import { UserIcon } from "@heroicons/react/24/outline";

import { RepositoryIssueCommentsQuery } from "@app/generated/graphql";

type Props = {
  comment: NonNullable<
    NonNullable<
      NonNullable<
        NonNullable<
          NonNullable<
            NonNullable<RepositoryIssueCommentsQuery["repository"]>["issue"]
          >["comments"]
        >["edges"]
      >[number]
    >["node"]
  >;
};

export const IssueComment = ({ comment }: Props) => {
  return (
    <div className="flex flex-row gap-2 items-start">
      <div className="flex flex-col rounded-full basis-6">
        <UserIcon className="h-6 w-6" />
      </div>
      <div className="flex flex-col grow-0 shrink-1">
        <p className="text-base font-bold">{comment.author?.login}</p>
        <p className="text-base mt-2">{comment.body}</p>
      </div>
    </div>
  );
};
