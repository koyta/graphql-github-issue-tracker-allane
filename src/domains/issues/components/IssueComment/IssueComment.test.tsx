import { findByText, render } from "@testing-library/react";

import { IssueComment } from "@app/domains/issues/components/IssueComment/IssueComment";

describe("IssueComment", () => {
  const comment = {
    body: "body",
    author: {
      login: "author",
    },
  };

  it("should render a comment body", async () => {
    const { container } = render(<IssueComment comment={comment as any} />);
    expect(await findByText(container, comment.body)).toHaveTextContent(
      comment.body
    );
  });
  it("should render a comment author", async () => {
    const { container } = render(<IssueComment comment={comment as any} />);
    expect(await findByText(container, comment.author.login)).toHaveTextContent(
      comment.author.login
    );
  });
});
