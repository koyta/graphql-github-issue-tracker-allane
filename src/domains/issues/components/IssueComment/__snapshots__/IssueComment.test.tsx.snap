// Vitest Snapshot v1

exports[`IssueComment > should render a comment 1`] = `
<div>
  <div
    class="flex flex-row gap-2 items-start"
  >
    <div
      class="flex flex-col rounded-full basis-6"
    >
      <svg
        aria-hidden="true"
        class="h-6 w-6"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.75 6a3.75 3.75 0 11-7.5 0 3.75 3.75 0 017.5 0zM4.501 20.118a7.5 7.5 0 0114.998 0A17.933 17.933 0 0112 21.75c-2.676 0-5.216-.584-7.499-1.632z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </div>
    <div
      class="flex flex-col grow-0 shrink-1"
    >
      <p
        class="text-base font-bold"
      >
        author
      </p>
      <p
        class="text-base mt-2"
      >
        body
      </p>
    </div>
  </div>
</div>
`;
